pkg/virt-controller/watch/clone/clone.go

```diff
@@ -359,9 +359,13 @@ func (ctrl *VMCloneController) createSnapshotFromVm(vmClone *clonev1alpha1.Virtu
 	snapshot := generateSnapshot(vmClone, vm)
 	syncInfo.logger.Infof("creating snapshot %s for clone %s", snapshot.Name, vmClone.Name)
 
-	snapshot, syncInfo.err = ctrl.client.VirtualMachineSnapshot(snapshot.Namespace).Create(context.Background(), snapshot, v1.CreateOptions{})
-	if syncInfo.err != nil {
-		return snapshot, addErrorToSyncInfo(syncInfo, fmt.Errorf("failed creating snapshot %s for clone %s: %v", snapshot.Name, vmClone.Name, syncInfo.err))
+	snapshot, err := ctrl.client.VirtualMachineSnapshot(snapshot.Namespace).Create(context.Background(), snapshot, v1.CreateOptions{})
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return snapshot, addErrorToSyncInfo(syncInfo, fmt.Errorf("failed creating snapshot %s for clone %s: %v", snapshot.Name, vmClone.Name, err))
+		}
+		syncInfo.snapshotName = snapshot.Name
```

Comment:
@mhenriks HI, do you think we need to log obj has already been created there?

Reply:
I don't think so.